kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java

```diff
@@ -395,10 +394,12 @@ private void enableInterface(String deviceId, NetworkProperties properties, Devi
             this.nm.ActivateConnection(new DBusPath(connection.get().getObjectPath()),
                     new DBusPath(device.getObjectPath()), new DBusPath("/"));
         } else {
-            AddAndActivateConnectionTuple createdConnectionTuple = this.nm.AddAndActivateConnection(
-                    newConnectionSettings, new DBusPath(device.getObjectPath()), new DBusPath("/"));
+            Settings settings = this.dbusConnection.getRemoteObject(NM_BUS_NAME, NM_SETTINGS_BUS_PATH, Settings.class);
+            DBusPath createdConnectionPath = settings.AddConnection(newConnectionSettings);
+            this.nm.ActivateConnection(createdConnectionPath, new DBusPath(device.getObjectPath()), new DBusPath("/"));
```

Comment:
Should we catch the exception thrown here and go on with the remaining parts of the method if the connection is added successfully but cannot be activated for some reason?

Reply:
I thought about it and decided against it since it should be safe as it is right now.

If an exception is thrown on the connection activation I cannot proceed with the housekeeping (I don't know which connection to spare from the deletion) and the modem watchdog setup.

What do you think?

Reply:
I'm in doubt.. I wonder if the configuration logic should depend or not on whether the interfaces can be brought up successfully at the instant when the configuration is applied. for example I would enable the modem monitor logic even if cannot connect at the moment.. what happens if the ActivateConnection call is completely removed?

Reply:
> Should we catch the exception thrown here and go on with the remaining parts of the method if the connection is added successfully but cannot be activated for some reason?

Addressed in 6ad61b89cbda386ed5ca5667a709e8e1cd548b4a